### ISSUE TITLE
Rules proposal: one describe per file

### ISSUE THREAD

**Issue by alecxe** [lo1tuma/eslint-plugin-mocha#103]
The rationale and the use case we have is described here [Enforce one describe per file](http://stackoverflow.com/questions/39436530/enforce-one-describe-per-file).

To summarize - the rule should produce a warning if there is more than one top-level `describe` per file.

Probably, we have too narrow of a use case - feel free to close if you feel there is no need for this kind of rule. Or, may be we can just turn it off by default.

Thanks!

**Comment by lo1tuma** [comment]
Hi @alecxe, thanks for the suggestion. I like the idea and would probably use such a rule as I’m already restrict my test files to only have one top-level describe 😉 .

Just for completeness: there are of course some limitations to static analysis in JavaScript where not all top-level describes can be detected. For example when someone creates test suites dynamically:

```js
function createTestCases(name) {
    describe(name, function () {
        // tests ...
    });
}

createTestCases('foo');
createTestCases('bar');
```

**Comment by jfmengels** [comment]
FYI, this is kind of similar to https://github.com/lo1tuma/eslint-plugin-mocha/issues/83.

**Comment by lo1tuma** [comment]
@jfmengels Good hint. I guess this rule together with a `no-nested-describes` rule would give you the behavior that you can only use one `describe` per file.

**Comment by jfmengels** [comment]
Ah, good point. Fine by me then :+1:
